hdate/tekufot.py

```diff
@@ -0,0 +1,256 @@
+        else:
+            date_equinox_april = dt.date(self.gregorian_year_p, 4, 8)
+
+        # Hours shift depends on leap year cycles
+        hours_delta_nissan = (self.gregorian_year_p % 4) * 6
+
+        # Tekufa Nissan: start at date_equinox_april at 12:00
+        tekufa_nissan = dt.datetime.combine(
+            date_equinox_april, dt.time(12, 0)
+        ) + dt.timedelta(hours=hours_delta_nissan)
+        self.tekufa_nissan = tekufa_nissan
+
+        # Tekufa intervals are about 91 days and 7.5 hours apart
+        tekufa_delta = dt.timedelta(
+            days=interval_days,
+            hours=int(interval_hours),
+            minutes=int((interval_hours - int(interval_hours)) * 60),
+        )
+
+        # From Nissan to Tevet (minus interval)
+        self.tekufa_tevet = self.tekufa_nissan - tekufa_delta
+        self.tekufa_tishrei = self.tekufa_tevet - tekufa_delta
+        self.tekufa_tammuz = self.tekufa_nissan + tekufa_delta
+
+    def get_cheilat_geshamim(self) -> dt.date:
+        """
+        Calculates the start date for the prayers for rain (Cheilat Geshamim).
+        In the diaspora, it is 60 days (add 59 days) after Tekufat Tishrei.
+        In Israel, it is fixed at the 7th of Cheshvan.
+        """
+
+        # Ensure we have Tekufa Tishrei
+        # if not hasattr(self, "tekufa_tishrei") or self.tekufa_tishrei is None:
+        #    self.cheilat_geshamim = None
+        #    return
+
+        if self.location.diaspora:
+            # Cheilat Geshamim starts 60 days after Tekufat Tishrei.
+            _cheilat_geshamim = self.tekufa_tishrei + dt.timedelta(days=59)
+
+            time_end_of_day = Zmanim(
+                _cheilat_geshamim.date(), location=self.location
+            ).zmanim["first_stars"]
+
+            tz = time_end_of_day.tzinfo
+
+            cheilat_geshamim_dt = dt.datetime(
+                _cheilat_geshamim.year,
+                _cheilat_geshamim.month,
+                _cheilat_geshamim.day,
+                _cheilat_geshamim.hour,
+                _cheilat_geshamim.minute,
+                tzinfo=tz,
+            )
+
+            if cheilat_geshamim_dt < time_end_of_day:
+                # Normalize to date at midnight
+                cheilat_geshamim = cheilat_geshamim_dt.date()
+            else:
+                cheilat_geshamim = cheilat_geshamim_dt.date() + dt.timedelta(days=1)
+        else:
+            # In Israel: 7th of Cheshvan
+            hdate_7_cheshvan = HebrewDate(self.hebrew_year_p, 2, 7)
+            jdn_7_cheshvan = conv.hdate_to_jdn(hdate_7_cheshvan)
+            cheilat_geshamim = conv.jdn_to_gdate(jdn_7_cheshvan)
+
+        return cheilat_geshamim
+
+    def get_cheilat_geshamim_hdate(self) -> HebrewDate:
+        """
+        Convert Cheilat Geshamim in Hebrew date.
+        """
+        jdn_geshamin = conv.gdate_to_jdn(self.get_cheilat_geshamim())
+
+        return conv.jdn_to_hdate(jdn_geshamin)
+
+    def get_gevurot(self) -> Gevurot:
```

Comment:
Why do we need both pesach-shemin atzeret of this year and previous year?
If we're dealing with a specific year we shouldn't need both.